src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,327 @@
+            false);
+
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+get_service_state(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    return state_dispatch(node, data);
+}
+
+static int
+set_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    const char *in_value;
+    int r;
+
+    r = sol_flow_packet_get_string(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    sol_util_replace_str_if_changed(&mdata->service_name, in_value);
+    SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+
+    return state_dispatch(node, data);
+}
+
+static int
+open_network_service(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct network_service_data *mdata = data;
+    const struct sol_flow_node_type_netctl_service_options *opts;
+    int r;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
+        SOL_FLOW_NODE_TYPE_NETCTL_SERVICE_OPTIONS_API_VERSION,
+        -EINVAL);
+    opts = (const struct sol_flow_node_type_netctl_service_options *)
+        options;
+
+    if (opts->name) {
+        mdata->service_name = strdup(opts->name);
+        SOL_NULL_CHECK(mdata->service_name, -ENOMEM);
+    }
+
+    mdata->state = SOL_NETCTL_SERVICE_STATE_UNKNOWN;
+
+    r = sol_netctl_add_service_monitor(service_status_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_netctl_add_error_monitor(error_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return state_dispatch(node, data);
+}
+
+static void
+close_network_service(struct sol_flow_node *node, void *data)
+{
+    struct network_service_data *mdata = data;
+
+    free(mdata->service_name);
+
+    if (sol_netctl_del_service_monitor(service_status_cb, node))
+        SOL_WRN("Could not delete service monitor !");
+
+    if (sol_netctl_del_error_monitor(error_cb, node))
+        SOL_WRN("Could not delete error monitor !");
+}
+
+void
+manager_cb(void *data)
+{
+    struct sol_flow_node *node = data;
+    bool offline;
+    int r;
+
+    offline = sol_netctl_get_radios_offline();
+
+    r = sol_flow_send_bool_packet(node,
+        SOL_FLOW_NODE_TYPE_NETCTL_RADIO_OFFLINE__OUT__OUT,
+        offline);
+    if (r < 0)
+        SOL_WRN("Failed to send radio offline status !");
+}
+
+static int
+enable_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    bool value;
+    int r;
+
+    r = sol_flow_packet_get_bool(packet, &value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_netctl_set_radios_offline(value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+open_network_radio(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+
+    r = sol_netctl_add_manager_monitor(manager_cb, node);
+    SOL_INT_CHECK(r, < 0, r);
```

Comment:
like I wrote for other node type, here you should call `manager_cb()` so it will report the current status (online/offline).